vapi/compute/policy.go

```diff
@@ -0,0 +1,133 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package compute
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"path"
+
+	"github.com/vmware/govmomi/vapi/compute/internal"
+	"github.com/vmware/govmomi/vapi/rest"
+)
+
+// PolicyManager provides convenience methods to operate Compute Policy
+type PolicyManager struct {
+	*rest.Client
+}
+
+// NewPolicyManager creates a new PolicyManager with the given client
+func NewPolicyManager(client *rest.Client) *PolicyManager {
+	return &PolicyManager{
+		Client: client,
+	}
+}
+
+// Policy represents the structure a compute policy.
+type Policy struct {
+	Capability  string `json:"capability,omitempty"`
+	Description string `json:"description,omitempty"`
+	Name        string `json:"name,omitempty"`
+	Policy      string `json:"policy,omitempty"`
+}
+
+// PolicySpec represents the structure to create a compute policy.
+type PolicySpec struct {
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	Capability  string `json:"capability,omitempty"`
+	HostTag     string `json:"host_tag,omitempty"`
+	VMTag       string `json:"vm_tag,omitempty"`
+}
+
+// CapabilitySummary
+type CapabilitySummary struct {
+	Capability  string `json:"capability,omitempty"`
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+}
+
+// Create creates a new Compute Policy and returns the id.
+func (c *PolicyManager) Create(ctx context.Context, policy PolicySpec) (string, error) {
+	spec := struct {
+		PolicySpec `json:"spec"`
+	}{policy}
+
+	r := c.Resource(internal.PolicyPath)
+	var res string
+	return res, c.Do(ctx, r.Request(http.MethodPost, spec), &res)
+}
+
+// Delete deletes a Compute Policy.
+func (c *PolicyManager) Delete(ctx context.Context, id string) error {
+	r := c.Resource(path.Join(internal.PolicyPath, id))
+	return c.Do(ctx, r.Request(http.MethodDelete), nil)
+}
+
+// List returns information about the compute policies available in this vCenter server.
+func (c *PolicyManager) List(ctx context.Context) ([]Policy, error) {
+	r := c.Resource(internal.PolicyPath)
+	var res []Policy
+	return res, c.Do(ctx, r.Request(http.MethodGet), &res)
+}
+
+// Get returns information about a specific compute policy
+func (c *PolicyManager) Get(ctx context.Context, id string) ([]Policy, error) {
+	r := c.Resource(path.Join(internal.PolicyPath, id))
+	var res []Policy
+	return res, c.Do(ctx, r.Request(http.MethodGet), &res)
+}
+
+// ListCapabilities returns information about the compute policy capabilities available in this vCenter server.
+func (c *PolicyManager) ListCapabilities(ctx context.Context) ([]CapabilitySummary, error) {
+	// Name+Description fields are of type LocalizableMessage in 6.9, but string in 7.0
```

Comment:
@shylasrinivas with this change, ListCapabilities works with both 6.9 and 7.0

Reply:
@dougm thanks for making this change. We just had a team discussion and decided to support only 7.0 APIs in terraform. With that in mind, it is ok if these 2 fields are strings.